### PR TITLE
fix: memory leaks in tmux windows

### PR DESCRIPTION
**Pull request by wv-tud**
## Summary
Improved cleanup of orphaned terminals and proper session deletion to prevent resource leaks when tmux windows are closed or sessions are deleted.

## Changes
- Added `window_exists()` method to TmuxClient for efficient window checking
- Added `kill_window()` method to properly close tmux windows
- Enhanced session deletion with verification and comprehensive cleanup
- Added grace period (1 hour) before deleting orphaned terminals in cleanup service
- Improved terminal deletion to clean up tmux windows and log files
- Auto-cleanup of orphaned terminal records when tmux windows no longer exist
- Enhanced `get_session()` to filter out terminals with missing tmux windows
- Enhanced `get_terminal()` to detect and clean up missing tmux windows

## Tests
- Added 49 comprehensive tests covering all critical functionality
- **New test file**: `test/clients/test_tmux_window_management.py` (40 tests)
  - 100% coverage of `window_exists()` and `kill_window()` methods
  - Tests for all error handling paths and edge cases
  - Session and window lifecycle management tests
- **Updated**: `test/services/test_cleanup_orphaned_terminals.py` (9 tests)
  - Tests cover orphaned terminal detection, grace period logic, and tmux errors
- Fixed 11 existing tests to match new implementation

## Coverage Improvements
- tmux.py: 19% → 59% (critical methods at 100%)
- session_service.py: 62% → 83%
- terminal_service.py: 81% → 82%
- cleanup_service.py: 96%

## Impact
This prevents memory leaks from accumulating when:
1. Users manually close tmux windows outside of CAO
2. Sessions are deleted but cleanup fails partially
3. Tmux crashes or is restarted leaving orphaned database records

### CODE REVIEW

File: coverage2.json
Comment (haofeif) [inline coverage2.json]: coverage.json and coverage2.json committed — These are generated files (~95KB each) that should not be tracked in the repo. Please remove them and add coverage`*.json` to `.gitignore`.

**Review by haofeif — CHANGES_REQUESTED** [review]
Also can you pls help to update the documents related, and also fix the unit testing issue ?

File: src/cli_agent_orchestrator/services/terminal_service.py [hunk 1]
```diff
@@ -145,28 +145,50 @@ def create_terminal(
...
         if not metadata:
             raise ValueError(f"Terminal '{terminal_id}' not found")
 
-        # Get status from provider
+        # Check if tmux window still exists
+        if not tmux_client.window_exists(metadata["tmux_session"], metadata["tmux_window"]):
+            logger.info(f"Cleaning up terminal {terminal_id} - tmux window no longer exists")
+            from cli_agent_orchestrator.clients.database import delete_terminal
+
+            delete_terminal(terminal_id)
```
Comment (haofeif) [inline src/cli_agent_orchestrator/services/terminal_service.py]: This imports the DB-only delete_terminal from database.py, which only deletes the DB record. It should call the local `delete_terminal()` (defined at line 331 in the same file) which does full 5-step cleanup:
* stop pipe-pane
* cleanup provider
* kill tmux window
* delete DB record
* delete log file. 
As written, this leaks the provider object in memory, leaves the log file on disk, and doesn't kill the tmux window

File: src/cli_agent_orchestrator/services/session_service.py [hunk 2]
```diff
@@ -44,42 +44,87 @@ def list_sessions() -> List[Dict]:
...
+        # Get session from tmux
+        session_data = tmux_client.get_session(session_name)
         if not session_data:
+            # Session doesn't exist - clean up any orphaned terminal records
+            terminals = list_terminals_by_session(session_name)
+            if terminals:
+                logger.info(
+                    f"Cleaning up {len(terminals)} orphaned terminals for non-existent session {session_name}"
+                )
+                delete_terminals_by_session(session_name)
```
Comment (haofeif) [inline src/cli_agent_orchestrator/services/session_service.py]: `get_session()` is a read operation that unexpectedly deletes DB records as a side effect. Should be a separate cleanup function or at minimum renamed to make the mutation explicit.

File: src/cli_agent_orchestrator/services/cleanup_service.py [hunk 3]
```diff
@@ -26,6 +29,38 @@ def cleanup_old_data():
...
+                    continue
+
+                # Check if tmux window still exists
+                try:
+                    if not tmux_client.window_exists(terminal.tmux_session, terminal.tmux_window):
+                        logger.info(
+                            f"Cleaning up orphaned terminal {terminal.id} - "
+                            f"tmux window no longer exists (inactive since {terminal.last_active})"
+                        )
+                        delete_terminal(terminal.id)
```
Comment (haofeif) [inline src/cli_agent_orchestrator/services/cleanup_service.py]: The outer session holds terminal ORM objects while delete_terminal opens a separate `SessionLocal()`. With SQLite this can cause database is locked errors.
Fix: collect IDs first, close the session, then delete in a separate loop.

**Review by haofeif — COMMENTED** [review]
A few more comments

**Comment by haofeif** [comment]
close it as no traction for 3 months.
